### ISSUE TITLE
Support named pointcuts

### ISSUE THREAD

**Issue by kdambekalns** [AOP-PHP/AOP#19]
For pointcut expressions that are complex or used often, a named pointcut is very helpful. Here is how FLOW3 does it: http://flow3.typo3.org/documentation/guide/partiii/aspectorientedprogramming.html#pointcuts

**Comment by geraldcroes** [comment]
I may have overlooked the named pointcut concepts, but how is it different from a simple constant ?

``` php
    <?php
    //define pointcut for all delete methods
    define('POINTCUT_DELETE', '**/*->delete*()');
    aop_add_before(POINTCUT_DELETE, $closure);
```

**Comment by kdambekalns** [comment]
True, on the language level of your extension that is about the same. We could have done that as well, I guess.

But when you combine named pointcuts with another expression, it becomes a little messy:

``` PHP
<?php
define('POINTCUT_DELETE', '**/*->delete*()');
aop_add_before('(Foo->bar() && ' . POINTCUT_DELETE . ') || Baz->Quux()' , $closure);
```

The other thing is, it pollutes the constant namespace - but only for a given PHP namespace, so it might not be that bad.

**Comment by geraldcroes** [comment]
Ok, I now understand your point.

We could as you propose include some kind of a pre-parse for selectors like

``` php
   <?php
    aop_pointcut_define('**/*->delete*()', 'pointcut_delete');
    aop_add_before('(Foo->bar() && pointcut_delete ) || Baz->Quux()' , $closure);
```

Maybe more then 

``` php
   <?php
    aop_pointcut_define('**/*->delete*()', 'delete');
    aop_add_before('(method(Foo->bar()) && pointcut(delete) ) || method(Baz->Quux())' , $closure);
```

The global syntax is really to be think of. I'm really the kind of person who prefer simplicity over completeness in API, letting the user define his own "what the base syntax cannot do".

I think we'll have to wait for more feedback and decide the way to go with this.
